### PR TITLE
Fix flaky stopping simultaneous datastream test

### PR DESCRIPTION
**Pull request by shrinandthakkar**
Note: The timeout of 100ms turned out to be a little less for the test to finish, so instead of increasing the timeout there, added a polling loop, which will poll every 100ms for 1 sec to check the status of the datastream after the execution of the test. 

Important: DO NOT REPORT SECURITY ISSUES DIRECTLY ON GITHUB.  
For reporting security issues and contributing security fixes,  
please, email security@linkedin.com instead, as described in  
the contribution guidelines.

Please, take a minute to review the contribution guidelines at:  
https://github.com/linkedin/Brooklin/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by somandal — APPROVED** [review]
lgtm

**Comment by surajkn** [comment]
Please add a note in commit message to mention why the test was flaky or what you did to fix it.

File: datastream-server-restli/src/test/java/com/linkedin/datastream/server/dms/TestDatastreamResources.java [hunk 1]
```diff
@@ -296,24 +295,16 @@ public void testSimultaneousStopDatastreamRequests() {
...
-    } finally {
-      executor.shutdownNow();
-    }
+    executor.execute(() -> resource2.stop(pathKey, false));
 
     // Retrieve datastream and check that is in STOPPED state.
-    Datastream ds = resource2.get(datastreamName);
-    Assert.assertNotNull(ds);
-    Assert.assertEquals(ds.getStatus(), DatastreamStatus.STOPPED);
+    Assert.assertTrue(PollUtils.poll(() -> resource1.get(datastreamName).getStatus() == DatastreamStatus.STOPPED, 100, 1000));
```
Comment (surajkn) [inline datastream-server-restli/src/test/java/com/linkedin/datastream/server/dms/TestDatastreamResources.java:306]: Do we not need to check for resource2 status ?

Comment (shrinandthakkar) [reply datastream-server-restli/src/test/java/com/linkedin/datastream/server/dms/TestDatastreamResources.java:306]: nope, they are just resource endpoints, both of them would give the same output. We just have two of them in the test to mock the scenario of having multiple endpoints requesting the status change.

**Comment by surajkn** [comment]
lgtm

**Review by surajkn — APPROVED** [review]

**Pull request merged**